modules/openapi-generator/src/main/resources/typescript-angular/api.base.service.mustache

```diff
@@ -28,21 +28,30 @@ export class BaseService {
         return consumes.indexOf('multipart/form-data') !== -1;
     }
 
-    protected addToHttpParams(httpParams: HttpParams, value: any, key?: string): HttpParams {
+    protected addToHttpParams(httpParams: HttpParams, value: any, key?: string, isDeep?: boolean): HttpParams {
         // If the value is an object (but not a Date), recursively add its keys.
         if (typeof value === 'object' && !(value instanceof Date)) {
+            if (isDeep) {
+                return this.addToHttpParamsRecursive(httpParams, value, key, isDeep);
+            }
             return this.addToHttpParamsRecursive(httpParams, value);
```

Comment:
```suggestion
        return this.addToHttpParamsRecursive(httpParams, value, key, isDeep);
```

should be enough with a default `key` of `undefined` and `isDeep = false`?

Reply:
Not sure, key was explicitly removed here: 8ead4b6e3f18bcc437a80fe8df68e5be7a945f8d as part of #20850.
So one of the codepath seems to need the key unset while another needs it set.
If the verbosity is the issue, I'm happy to go with a ternary operator instead:
```suggestion
            return this.addToHttpParamsRecursive(httpParams, value, isDeep ? key : undefined, isDeep);
```

Reply:
Added ternary, see 810f2ad1616c07e4a5673f7438b828d53594e7ec.